checker/watcher.go

```diff
@@ -169,16 +173,21 @@ func (pw *PeersWatcher) doWork() {
 
 	// Figure out current connectivity state
 	ok := true
-	for peerIP, peer := range pw.peersMapByIP {
-		if !peer.Consider() {
-			log.Debugf("Peer(%v): not considered for connectivity state", peer.uuid)
-			continue
-		}
-		log.Debugf("peer(%v): %+v", peerIP, peer)
-		if peer.count == 0 {
-			ok = false
-			log.Debugf("peer: %v is not reachable", peerIP)
+	if shouldConsider(mdInfo) {
+		for peerIP, peer := range pw.peersMapByIP {
+			if !peer.Consider() {
+				log.Debugf("Peer(%v): not considered for connectivity state", peer.uuid)
+				continue
+			}
+			log.Debugf("peer(%v): %+v", peerIP, peer)
+			if peer.count == 0 {
+				ok = false
```

Comment:
should we break here? once ok is set to false

Reply:
yeah true, but left it that way to get debug logs to figure out all the peers which have problems

Reply:
sounds good